src/lib.rs

```diff
@@ -276,6 +293,29 @@ pub fn store<'a, T: Serialize>(
 ///
 /// [`store`]: fn.store.html
 pub fn store_path<T: Serialize>(path: impl AsRef<Path>, cfg: T) -> Result<(), ConfyError> {
+    _store(path, cfg, None)
+}
+
+/// Save changes made to a configuration object at a specified path
+///
+/// This is an alternate version of [`store_path`] that allows the
+/// specification of file permissions that must be set. For more information on
+/// errors and behavior, see [`store`]'s documentation.
+///
+/// [`store_path`]: fn.store_path.html
+pub fn store_path_perms<T: Serialize>(
+    path: impl AsRef<Path>,
+    cfg: T,
+    perms: Permissions,
+) -> Result<(), ConfyError> {
+    _store(path, cfg, Some(perms))
+}
+
+fn _store<T: Serialize>(
+    path: impl AsRef<Path>,
```

Comment:
Since the function is private, we can avoid compilation time/bloat by removing the generic argument. The public functions should transform the generic arguments into concrete ones.
```suggestion
    path: &Path,
```